tests/managedmcg_controller_test.go

```diff
@@ -6,45 +6,189 @@ import (
 	. "github.com/onsi/gomega"
 	managedmcgv1alpha1 "github.com/red-hat-storage/mcg-osd-deployer/api/v1alpha1"
+	"github.com/red-hat-storage/mcg-osd-deployer/controllers"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	noobaav1alpha1 "github.com/noobaa/noobaa-operator/v5/pkg/apis/noobaa/v1alpha1"
+	opv1a1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
 const (
-	Namespace = "openshift-storage"
+	namespace = "redhat-data-federation"
 )
 
 var (
 	managedMCGName = "managedmcg"
 	managedMCG     = managedmcgv1alpha1.ManagedMCG{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      managedMCGName,
-			Namespace: Namespace,
+			Namespace: namespace,
 		},
 		Spec: managedmcgv1alpha1.ManagedMCGSpec{
 			ReconcileStrategy: "ignore",
 		},
 	}
+	ns = &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		},
+	}
+	r = &controllers.ManagedMCGReconciler{
+		Log: ctrl.Log.WithName("controllers").WithName("ManagedMCGFake"),
+	}
 )
 
 var _ = Describe("ManagedMCG validations", func() {
+
 	BeforeEach(func() {
-		namespace := &corev1.Namespace{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: Namespace,
-			},
-		}
-		err := k8sClient.Create(context.Background(), namespace, &client.CreateOptions{})
-		Expect(err).NotTo(HaveOccurred())
+		r.Scheme = k8sClient.Scheme()
 	})
+
 	When("Creating and deleting ManagedMCG", func() {
 		It("should not return validation error", func() {
 			By("using default values", func() {
+				newNamespace := ns.DeepCopy()
 				newManagedMCG := managedMCG.DeepCopy()
-				err := k8sClient.Create(context.Background(), newManagedMCG, &client.CreateOptions{})
+				r.Client = fake.NewClientBuilder().WithScheme(k8sClient.Scheme()).WithObjects(newNamespace, newManagedMCG).Build()
+
+				err := r.Client.Get(context.Background(), client.ObjectKeyFromObject(newManagedMCG), newManagedMCG)
+				Expect(err).NotTo(HaveOccurred())
+
+				err = r.Client.Delete(context.Background(), newManagedMCG, &client.DeleteOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				err = r.Client.Get(context.Background(), client.ObjectKeyFromObject(newNamespace), newNamespace)
+				Expect(err).NotTo(HaveOccurred())
+
+				err = r.Client.Delete(context.Background(), newNamespace, &client.DeleteOptions{})
 				Expect(err).NotTo(HaveOccurred())
-				err = k8sClient.Delete(context.Background(), newManagedMCG, &client.DeleteOptions{})
+			})
+		})
+	})
+})
+
+var _ = Describe("ManagedMCGReconciler Reconcile", func() {
+	When("Creating ManagedMCG", func() {
+		newNamespace := ns.DeepCopy()
+		newManagedMCG := managedMCG.DeepCopy()
+
+		addonSecretFake := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "addon-secret-fake",
+				Namespace: namespace,
+			},
+		}
+		addonSecretFake.Data = make(map[string][]byte, 1)
```

Comment:
just like ObjectMeta can you add Data also inside the struct

Reply:
okay @naveenpaul1 , I will do that. Thanks!